AndroidAnnotations/androidannotations-core/androidannotations-test/src/main/java/org/androidannotations/test/eview/ViewWithInstanceState.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (C) 2016 the AndroidAnnotations project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed To in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.androidannotations.test.eview;
+
+import org.androidannotations.annotations.EView;
+import org.androidannotations.annotations.InstanceState;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.View;
+
+@EView
+public class ViewWithInstanceState extends View {
```

Comment:
I think we should have another test, which asserts we correctly retain the saved instance information from the super class. So there should be a non-AA Android custom View class, which have `onSaveInstanceState()` and `onRestoreInstanceState()`, then an enhanced class should subclass it, and add `@InstanceState`. We should assert that the state restoration in parent works as well.